escape textareas adjustment for v.long content

i-just opened the pull request:
### Description
Adjustment for `Html::_escapeTextareas()` to handle `$html` that’s long enough to trigger `pcre.backtrack_limit` in `preg_replace_callback()`.


### Related issues
#13083

brandonkelly commented on src/helpers/Html.php:
```diff
@@ -760,11 +760,27 @@ function(array $match) use ($namespace, $ids): string {
+            /* @phpstan-var $data string */
+            $html = $data;
+        } else {
+            preg_match('/(.*?)(<textarea\b[^>]*>)/is', $html, $matches);
+            $start = $matches[0];
+            preg_match('/(<\/textarea>.*)/is', $html, $matches);
+            $end = $matches[0];
+            $marker = '{marker:' . StringHelper::randomString() . '}';
+            $markers[$marker] = str_replace([$start, $end], '', $html);
+            $html = $start . $marker . $end;
```
This looks like it’s assuming there is only 1 `<textarea>` tag. Think you’d need to keep doing this in a loop until `preg_match` returns `0`.

And if this method is more performant than `preg_replace_callback()`, we should probably just go straight to using it, rather than just as a fallback if `$data` is` null`.

i-just replied:
I adjusted it for multiple `<textarea>` tags and dropped the `preg_replace_callback()` entirely.

Tested against "more then 1 million chars" as metioned in the issue and also against 16MB of text in a field and this version is still able to process it all.

i-just commented:
I adjusted it for multiple `<textarea>` tags and dropped the `preg_replace_callback()` entirely.

Tested against "more then 1 million chars" as metioned in the issue and also against 16MB of text in a field and this version is still able to process it all.
